src/server/pegasus_hotkey_collector.h

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <functional>
+#include <dsn/utility/error_code.h>
+#include <rrdb/rrdb_types.h>
+#include <gtest/gtest_prod.h>
+#include <readerwriterqueue/readerwriterqueue.h>
+#include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/flags.h>
+#include "base/pegasus_utils.h"
+
+namespace pegasus {
+namespace server {
+
+DSN_DECLARE_int32(data_capture_hash_bucket_num);
+
+typedef std::shared_ptr<std::unordered_map<int, int>> thread_queue_map;
+
+class hotkey_coarse_data_collector;
+class hotkey_fine_data_collector;
+
+enum class collector_state
+{
+    STOP,   // data has been cleared, ready to start
+    COARSE, // is running corase capture and analyse
+    FINE,   // is running fine capture and analyse
+    FINISH  // capture and analyse is done, ready to get result
+};
+
+class hotkey_collector
```

Comment:
```suggestion
class hotkey_collector : public replica_base
```

If this is a replica-level class, define it as a `replica_base`, which means it is one-instance-per-replica.
You can remove `replica_name()` and `_replica_name` after using replica_base.